### PR TITLE
feat(protobuf): WIP-0020 Support

### PR DESCRIPTION
**Pull request by parodyBit**
extended RADType Enum to support HttpPost and RNG
added StringPair to Schema
refactor to_pb_bytes methods to return correct bytes

added tests to verify correct bytecode

### CODE REVIEW

**Review by parodyBit — COMMENTED** [review]
@drcpu-github

File: util/protobuf_encoder.py [hunk 1]
```diff
@@ -101,25 +115,86 @@ def from_json(cls, data):
...
+
+            string_pairs = []
+            for key in data["headers"]:
+                string_pairs.append(StringPair(left=key, right=data["headers"][key]))
+            return RADRetrieve(
+                kind=RADType.from_json(data['kind']),
+                url=data['url'],
+                script=bytes(data['script']),
+                body=bytes(data['body']),
+                headers=string_pairs)
```
Comment (drcpu-github) [inline util/protobuf_encoder.py:158]: For the sake of formatting, please put the closing parenthesis on a new line (and aligned with `return`)

File: util/protobuf_encoder.py [hunk 2]
```diff
@@ -101,22 +101,82 @@ def from_json(cls, data):
...
+    body: bytes
+    headers: List[StringPair]
 
     @classmethod
     def from_json(cls, data):
-        return RADRetrieve(kind=RADType.from_json(data['kind']), url=data['url'], script=bytes(data['script']))
+        _body: bytes = b''
+        if 'body' in data:
+            _body = bytes(data['body'])
+        if data['kind'] == 'HTTP-POST':
```
Comment (drcpu-github) [inline util/protobuf_encoder.py:131]: Probably nicer to handle this similarly to `_body` (so have an extra `headers=[]` before the if-statement) and unident the `return RADRetrieve` to have less duplication?

File: util/protobuf_encoder.py [hunk 3]
```diff
@@ -101,22 +101,82 @@ def from_json(cls, data):
...
+            body_bytes = pb_field(field_number=4, tag=LENGTH_DELIMITED, value=self.body)
+        else:
+            body_bytes = b''
+
+        if len(self.headers) > 0:
+            header_bytes = pb_field(field_number=5, tag=LENGTH_DELIMITED, value=concat([x.to_pb_bytes() for x in self.headers]))
+        else:
+            header_bytes = b''
+
+        if self.kind.value == 3:
```
Comment (drcpu-github) [inline util/protobuf_encoder.py:178]: Can we compare `RadType.HttpPost` for clarity?

**Review by drcpu-github — CHANGES_REQUESTED** [review]

File: util/protobuf_encoder.py (hunk 2 above)
Comment (parodyBit) [reply util/protobuf_encoder.py]: agreed, 
less duplication and uniform variable declarations

**Review by drcpu-github — APPROVED** [review]

**Pull request merged**